pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -103,38 +103,34 @@ func SetDefaults_Seed(obj *Seed) {
 	}
 
 	if obj.Spec.Settings.ExcessCapacityReservation == nil {
-		enabled := true
-		for _, taint := range obj.Spec.Taints {
-			if taint.Key == DeprecatedSeedTaintDisableCapacityReservation {
-				enabled = false
-			}
-		}
-		obj.Spec.Settings.ExcessCapacityReservation = &SeedSettingExcessCapacityReservation{Enabled: enabled}
+		obj.Spec.Settings.ExcessCapacityReservation = &SeedSettingExcessCapacityReservation{Enabled: true}
 	}
 
 	if obj.Spec.Settings.Scheduling == nil {
-		visible := true
-		for _, taint := range obj.Spec.Taints {
-			if taint.Key == DeprecatedSeedTaintInvisible {
-				visible = false
-			}
-		}
-		obj.Spec.Settings.Scheduling = &SeedSettingScheduling{Visible: visible}
+		obj.Spec.Settings.Scheduling = &SeedSettingScheduling{Visible: true}
 	}
 
 	if obj.Spec.Settings.ShootDNS == nil {
-		enabled := true
-		for _, taint := range obj.Spec.Taints {
-			if taint.Key == DeprecatedSeedTaintDisableDNS {
-				enabled = false
-			}
-		}
-		obj.Spec.Settings.ShootDNS = &SeedSettingShootDNS{Enabled: enabled}
+		obj.Spec.Settings.ShootDNS = &SeedSettingShootDNS{Enabled: true}
 	}
 
 	if obj.Spec.Settings.VerticalPodAutoscaler == nil {
 		obj.Spec.Settings.VerticalPodAutoscaler = &SeedSettingVerticalPodAutoscaler{Enabled: true}
 	}
+
+	taintsToRemove := []string{
+		"seed.gardener.cloud/disable-capacity-reservation",
+		"seed.gardener.cloud/disable-dns",
+		"seed.gardener.cloud/invisible",
+	}
+	// remove taints
+	for _, taint := range taintsToRemove {
+		for i := len(obj.Spec.Taints) - 1; i >= 0; i-- {
+			if obj.Spec.Taints[i].Key == taint {
+				obj.Spec.Taints = append(obj.Spec.Taints[:i], obj.Spec.Taints[i+1:]...)
+			}
+		}
+	}
```

Comment:
I am not fully confident that I have to forcefully remove the taints from the seed resource. Keeping them should not be a problem now when we no longer sync them with spec.settings fields. So any feedback will be appreciated.

Reply:
Well, I am fine with both options. If we pick this approach to remove them during defaulting, then I would suggest to add a TODO that reminds us to remove this migration code in a future version.
@gardener/gardener-maintainers , WDYT?

Reply:
I'm also fine with removing the taints here.

Reply:
After some time I think we need the above clean up for one minor release to prevent orphaning these 3 taints, otherwise shoots without respective toleration cannot be scheduled on the seed. Of course, we can skip the clean up but then a Gardener operator might need to manually clean up them.